src/screens/NoticeboardFormScreen.tsx

```diff
@@ -0,0 +1,490 @@
+import { StackNavigationProp } from '@react-navigation/stack';
+import moment from 'moment';
+import { extendMoment } from 'moment-range';
+import React, { useCallback, useContext, useState } from 'react';
+import { useMutation } from 'react-apollo';
+import { Control, Controller, DeepMap, FieldValues, useForm } from 'react-hook-form';
+import { ActivityIndicator, Alert, RefreshControl, ScrollView, StyleSheet } from 'react-native';
+
+import {
+  BoldText,
+  Button,
+  Checkbox,
+  DateTimeInput,
+  HtmlView,
+  Input,
+  LoadingContainer,
+  RegularText,
+  Title,
+  TitleContainer,
+  TitleShadow,
+  Touchable,
+  Wrapper,
+  WrapperRow
+} from '../components';
+import { colors, consts, device, texts } from '../config';
+import { momentFormat } from '../helpers';
+import { useStaticContent } from '../hooks';
+import { NetworkContext } from '../NetworkProvider';
+import { CREATE_GENERIC_ITEM, CREATE_GENERIC_ITEM_MESSAGE } from '../queries/genericItem';
+import { NoticeboardType } from '../types';
+
+const { EMAIL_REGEX } = consts;
+const a11yText = consts.a11yLabel;
+const extendedMoment = extendMoment(moment);
+
+type NoticeboardFormScreenDataType = {
+  contentBlocks: [{ body: string; title: string }];
+  createdAt?: string;
+  dates: [{ dateEnd: string; dateStart: string }];
+  id: string | number;
+  title: string;
+};
+
+type NoticeboardData = {
+  body: string;
+  dateEnd: string;
+  dateStart: string;
+  email: string;
+  message: string;
+  name: string;
+  noticeboardType: string;
+  phoneNumber: string;
+  termsOfService: boolean;
+  title: string;
+};
+
+const NOTICEBOARD_TYPE = [
+  { value: NoticeboardType.Offers, title: texts.noticeboard.offers },
+  { value: NoticeboardType.Searches, title: texts.noticeboard.searches },
+  { value: NoticeboardType.NeighbourlyHelp, title: texts.noticeboard.neighbourlyHelp }
+];
+
+const alert = (alertType: string) => {
+  let title = texts.noticeboard.alerts.hint,
+    message = '',
+    buttonText = texts.noticeboard.abort;
+
+  switch (alertType) {
+    case 'termsOfService':
+      message = texts.noticeboard.alerts.termsOfServiceMessage;
+      break;
+    case 'dateDifference':
+      message = texts.noticeboard.alerts.dateDifferenceMessage;
+      break;
+    case 'noticeboardType':
+      message = texts.noticeboard.alerts.noticeboardTypeMessage;
+      break;
+    case 'error':
+      message = texts.noticeboard.alerts.errorMessage;
+      break;
+    default:
+      title = texts.noticeboard.alerts.hint;
+      message = texts.noticeboard.alerts.errorMessage;
+      buttonText = texts.noticeboard.abort;
+      break;
+  }
+
+  return Alert.alert(title, message, [{ text: buttonText, style: 'cancel' }]);
```

Comment:
what about `message = texts.noticeboard.alerts.[alertType]` instead of switch case lines for?

Reply:
Wow, that would be absolutely perfect. 👍

Reply:
https://github.com/ikuseiGmbH/smart-village-app-app/pull/700/commits/6d57f01fbd1f7e857e1c42b5adad0cecca05471b